File: JavaScript/M03_JavaScriptProgrammingBasics/L03_ConditionalStatementsAdvanced/Lab/P13_SkiTrip.js
function skiTrip(input) {
    let days = parseInt(input[0]);
    let aptType = input[1];
    let rating = input[2];

    let stayLength = [days < 10, days >= 10 && days <= 15, days > 15];
    let nightPrice = 0.0, price = 0.0, discount = 0.0, totalPrice = 0.0, tip = 0.0;

    switch (aptType) {
        case "room for one person":
            nightPrice = 18.0;
            totalPrice = nightPrice * (days - 1);
            break;
        case "apartment":
            nightPrice = 25.0;
            price = nightPrice * (days - 1);

            if (stayLength[0]) {
                discount = (30 / 100) * price;
            } else if (stayLength[1]) {
                discount = (35 / 100) * price;
            } else if (stayLength[2]) {
                discount = (50 / 100) * price;
            }

            totalPrice = price - discount;
            break;
        case "president apartment":
            nightPrice = 35.0;
            price = nightPrice * (days - 1);

            if (stayLength[0]) {
                discount = (10 / 100) * price;
            } else if (stayLength[1]) {
                discount = (15 / 100) * price;
            } else if (stayLength[2]) {
                discount = (20 / 100) * price;
            }

            totalPrice = price - discount;
            break;
        default:
            break;
    }

    if (rating == "positive") {
        tip = (25 / 100) * totalPrice;
        totalPrice += tip;
    } else {
        totalPrice -= (10 / 100) * totalPrice;
    }

    console.log(totalPrice.toFixed(2));
}

File: Python/M01_ProgrammingBasics/L03_ConditionalStatementsAdvanced/Lab/Solutions/P13_SkiTrip.py
days = int(input())
room = input()
grade = input()

nights = days - 1
price = 0

if room == 'room for one person':
    price = nights * 18
elif room == 'apartment':
    if nights < 10:
        price = (nights * 25) * 0.7
    elif 10 <= nights < 15:
        price = (nights * 25) * 0.65
    else:
        price = (nights * 25) * 0.5

elif room == "president apartment":
    if nights < 10:
        price = (nights * 35) * 0.9
    elif 10 <= nights < 15:
        price = (nights * 35) * 0.85
    else:
        price = (nights * 35) * 0.8

if grade == 'positive':
    price = price * 1.25
elif grade == "negative":
    price = price * 0.9

print(f'{price:.2f}')

